### PR TITLE
a few type improvements, mostly flight related

### PR DESCRIPTION
**Pull request by dylanscott**
Hi, really appreciate your work on adding these stubs. I thought I'd open a PR with some minor additions/corrections encountered in my usage, mostly in the `pyarrow.flight` package, with one in `pyarrow.ipc`. I'll leave comments inline with sources.

### CODE REVIEW

File: pyarrow-stubs/__lib_pxi/ipc.pyi [hunk 1]
```diff
@@ -154,6 +154,8 @@ class _RecordBatchFileReader(_Weakrefable):
     def __enter__(self) -> Self: ...
     def __exit__(self, exc_type, exc_val, exc_tb): ...
     @property
+    def schema(self) -> Schema: ...
```
Comment (dylanscott) [inline pyarrow-stubs/__lib_pxi/ipc.pyi:157]: mentioned in [the docs](https://arrow.apache.org/docs/python/generated/pyarrow.ipc.RecordBatchFileReader.html#pyarrow.ipc.RecordBatchFileReader.schema) (though it's not explicit about the type) / [source link](https://github.com/apache/arrow/blob/6a2e19a852b367c72d7b12da4d104456491ed8b7/python/pyarrow/ipc.pxi#L1061-L1062) (with type)

File: pyarrow-stubs/_flight.pyi [hunk 2]
```diff
@@ -37,7 +37,7 @@ class CertKeyPair(NamedTuple):
     key: str
 
 class FlightError(Exception):
-    extra_info: str
+    extra_info: bytes
```
Comment (dylanscott) [inline pyarrow-stubs/_flight.pyi:40]: `bytes` type documented [here](https://arrow.apache.org/docs/python/generated/pyarrow.flight.FlightError.html)

File: pyarrow-stubs/_flight.pyi [hunk 3]
```diff
@@ -83,6 +83,9 @@ class BasicAuth(_Weakrefable):
     def username(self) -> bytes: ...
     @property
     def password(self) -> bytes: ...
+    def serialize(self) -> str: ...
+    @staticmethod
+    def deserialize(serialized: str | bytes) -> BasicAuth: ...
```
Comment (dylanscott) [inline pyarrow-stubs/_flight.pyi:88]: [source link](https://github.com/apache/arrow/blob/77f70e4a122c50f28d9ff97debb8831d2558f4d3/python/pyarrow/_flight.pyx#L413-L421) - the types aren't annotated but can be inferred. `deserialize` calls `tobytes` on its input like the constructor, and `serialize` calls `SerializeToString` (and `GetResultValue` appears to just move the value

File: pyarrow-stubs/_flight.pyi [hunk 4]
```diff
@@ -196,7 +199,7 @@ class _MetadataRecordBatchReader(_Weakrefable, _ReadPandasMixin):
 class MetadataRecordBatchReader(_MetadataRecordBatchReader): ...
 
 class FlightStreamReader(MetadataRecordBatchReader):
-    def cancel(self): ...
+    def cancel(self) -> None: ...
```
Comment (dylanscott) [inline pyarrow-stubs/_flight.pyi:202]: just missing a return annotation (and [no value is returned](https://github.com/apache/arrow/blob/77f70e4a122c50f28d9ff97debb8831d2558f4d3/python/pyarrow/_flight.pyx#L1065-L1068))

**Comment by zen-xu** [comment]
Thank you very much for providing these improvements!

**Pull request merged**